index.js

```diff
@@ -40,34 +40,35 @@ exports.register = (server, options) => {
 
   // catch request errors/warnings/etc (default: only errors) and capture them with sentry
   server.events.on({ name: 'request', channels: opts.channels }, async (request, event) => {
-    // format a sentry event from the request and triggered event
-    const sentryEvent = await Sentry.Parsers.parseError(event.error);
-    Sentry.Handlers.parseRequest(sentryEvent, request.raw.req);
+    Sentry.withScope(scope => { // thus use a temp scope and re-assign it
+      scope.addEventProcessor(_sentryEvent => {
+        // format a sentry event from the request and triggered event
+        const sentryEvent = Sentry.Handlers.parseRequest(_sentryEvent, request.raw.req);
```

Comment:
This method modifies the passed event. At least it did so in sentry@4.
So could we remove the assignment, just call the method and rename `_sentryEvent` to `sentryEvent` ?

Reply:
You are right, it modifies the original, so that assigning is not necessary. I still think that it makes sense to use return value as they do in their example and not rely on that its being just modified. That might change in future.  Reason for having _sentryEvent is just habit to avoid reassigning function arguments, which can have [performance impact](https://github.com/airbnb/javascript/issues/641#issuecomment-167827978), not sure if thats still the thing on latest node though.